ui/src/app/applications/components/application-details/application-resource-list.tsx

```diff
@@ -12,82 +12,98 @@ export const ApplicationResourceList = ({
     onNodeClick,
     nodeMenu
 }: {
-    resources: models.ResourceStatus[];
+    resources: any[];
     onNodeClick?: (fullName: string) => any;
     nodeMenu?: (node: models.ResourceNode) => React.ReactNode;
-}) => (
-    <div className='argo-table-list argo-table-list--clickable'>
-        <div className='argo-table-list__head'>
-            <div className='row'>
-                <div className='columns small-1 xxxlarge-1' />
-                <div className='columns small-2 xxxlarge-2'>NAME</div>
-                <div className='columns small-3 xxxlarge-4'>GROUP/KIND</div>
-                <div className='columns small-3 xxxlarge-3'>NAMESPACE</div>
-                <div className='columns small-2 xxxlarge-2'>STATUS</div>
+}) => {
+    const [showSyncOrder, setShowSyncOrder] = React.useState(false);
+    React.useEffect(() => {
+        for (const resource of resources) {
+            if (typeof resource.syncOrder !== 'undefined' && resource.syncOrder?.length !== 0) {
```

Comment:
i think it will be better change it to https://lodash.com/docs/#isEmpty

Reply:
Thanks for the recommendation, good learning 👍